docs/proposal-config-changes.md

```diff
@@ -0,0 +1,731 @@
+# Proposal - changes to global and workflow configs
+
+These notes mainly address [cylc-flow #3422](https://github.com/cylc/cylc-flow/issues/3422).
+
+## General points
+
+### Terminology
+   * We should remove all references to `suite` in the settings (partly because
+     we're trying stop referring to "suites" and partly because some affected
+     settings are actually referring to the workflow server which is unclear).
+     * Note: "suite" is referred to through-out the code, tests and
+       documentation.
+   * Do we want to use `server` as shorthand for `workflow server`?
+     * Presumably there is no confusion with `UI server` since this will have a
+       separate config file?
+     * Using `server` for now in this proposal.
+     * See
+       [discussion](https://github.com/cylc/cylc-flow/pull/3348#discussion_r332985001).
+
+### Deprecation
+   * The global config will be changing drastically for the platforms support.
+     Furthermore, it doesn't affect most end users.
+     * Proposal: Don't support any deprecated or obsolete settings in the global
+       config.
+   * For the workflow config the code to support deprecated and obsolete settings
+     isn't complicated so there is no big overhead for us. On the other hand we
+     don't want users using old syntax for too long. Some settings only became
+     deprecated or obsolete very recently - it seems harsh to remove support for
+     these.
+     * Proposal: Remove everything from 7.6.0 (Feb 2018) and before. That will
+       just leaves things from 7.8.x.
+
+### Config file naming
+
+See [cylc-flow #3170](https://github.com/cylc/cylc-flow/issues/3170).
+   * Suite: Currently `suite.rc` (specified in
+     [suite_files.py](https://github.com/cylc/cylc-flow/blob/master/cylc/flow/suite_files.py#L123)).
+     Since we're trying to move away from referring to "suites" we should change
+     this (with backwards compatibility).
+     * Proposal: `flow.rc` or `cylc-flow.rc`
+   * Global
+     * Currently `flow.rc` (specified in
+       [globalcfg.py](https://github.com/cylc/cylc-flow/blob/master/cylc/flow/cfgspec/globalcfg.py#L325)).
+       * Proposal: `config.rc`?
+     * Read from `/etc/cylc/flow/8.0a1/flow.rc` and
+       `~/.cylc/cylc/flow/8.0a1/flow.rc` unless `$CYLC_CONF_PATH` set in which
+       case it is read only from this path.
+     * We need the ability to override where the site config file is read from,
+       e.g. by setting `$CYLC_SITE_CONF_PATH`?
+     * Should we include the version in the path? Users should not have to
+       create a new config file every time we deploy a new version. Perhaps
+       major version only (with backwards compatibility assumed for minor
+       versions)?
```

Comment:
This would also require forward compatibility - which can be achieved by ignoring rather than failing unkown items. pros and cons...  (you do this with Rose configs already I think?)

Reply:
Not necessarily. You just can't use new features whilst still using old versions. We've lived with this until now.

Reply:
(I thought you were suggesting not have version-specific config files, in which case old cylc versions have to be able to handle newer config files ... which implies old cylc versions have to ignore - not fail - new or changed config items, doesn't it?)

Reply:
In Cylc7 we support version-specific OR non-version specific I believe.

I think going version specific makes the upgrade process unnecessarily difficult, particularly for independent installations.

We might be able to reduce the pain by being less specific e.g:

```
~/
  `- /.cylc/
     `- flow/
        |- 8/
        |  `- flow.rc
        |- 8.0/
        |  `- flow.rc
        `- 8.0.1/
           `- flow.rc
```

Reply:
> In Cylc7 we support version-specific OR non-version specific I believe.

Non version-specific?  Not really, because we fail on encountering unknown config items.  So it only works if new versions don't add new items.

Reply:
> **Non version-specific? Not really**

We do support version-specific config files at Cylc7:

https://github.com/cylc/cylc-flow/blob/128214a0f841d89694c76da67e458cd41e3363e7/lib/cylc/cfgspec/globalcfg.py#L376

Reply:
He heh, no - I was saying the we don't really support **non version-specific** config files.

We do support the non-specific file location, but if new cylc versions add new config items (and the new items are used in the file) it will break old versions of cylc that try to read the file.

Reply:
(Added @oliver-sanders comment post-merge) (the cylc7 discussion isn't relevant)